paasta_tools/kubernetes/remote_run.py

```diff
@@ -244,6 +284,45 @@ def remote_run_token(
     return create_temp_exec_token(kube_client, namespace, service_account)
 
 
+def generate_toolbox_deployment(
+    service: str, cluster: str, user: str
+) -> EksDeploymentConfig:
+    """Creates virtual EKS deployment for toolbox containers starting from adhoc configuration
+
+    :param str service: toolbox name
+    :param str cluster: target deployment cluster
+    :param str user: user requesting the toolbox
+    :return: deployment configuration
+    """
+    if not user.isalnum():
+        raise RemoteRunError(
+            f"Provided username contains non-alphanumeric characters: {user}"
+        )
+    adhoc_instance = service[len(TOOLBOX_MOCK_SERVICE) + 1 :]
```

Comment:
i think i'm missing something here - what will `service` look like here? from the callsites of this function, it looks like we would have `instance` available to pass in here)

Reply:
(i'm assuming that service as expected by this looks something like `prod-toolbox.instance` - but i'm also missing how that's getting setup)

Reply:
The change I proposed in soa-configs shows that

Reply:
ah gotcha - i think this further confuses me then: shouldn't `service` here be something just like `prod-toolbox` then? or are we munging the service name+instance into a single string at some point and i'm just missing where we're doing that

afaict, this is called from remote_run_state which includes the instance name - but we're not passing that through to here

Reply:
> are we munging the service name+instance into a single string

Users are specifying the munged name in full, we need that for authorization purposes. It's one of the reasons I had against using soa-configs vs system configurations, because I don't want to create N meta-services for N toolbox use cases.